### PR TITLE
GMG: code cleanup

### PR DESCRIPTION
**Pull request by tjhei**
- use make_shared
- rename variables
- formatting

### CODE REVIEW

File: source/simulator/stokes_matrix_free.cc [hunk 1]
```diff
@@ -256,7 +256,7 @@ namespace aspect
       // iterations of our two-stage outer GMRES iteration)
       if (do_solve_Schur_complement)
         {
-          // first solve with the bottom left block, which we have built
+          // first solve with the bottom right block, which we have built
```
Comment (MFraters) [inline source/simulator/stokes_matrix_free.cc:259]: hmm, that was probably just copied from here: https://github.com/geodynamics/aspect/blob/edcb0fa34bfdf4a5a49d58a90112837dfb95501e/source/simulator/solver.cc#L274, which also still has that. I guess that should be a different pull request.

**Review by MFraters — APPROVED** [review]
nice cleanup! Does it make sense to just write out mf as `matrix free` or `matrix_free_object`? It would make the code easier to read for me. Just a suggestion, fine either way.

File: source/simulator/stokes_matrix_free.cc (hunk 1 above)
Comment (tjhei) [reply source/simulator/stokes_matrix_free.cc:259]: true, see #5069

**Comment by tjhei** [comment]
> Does it make sense to just write out mf as `matrix free`

sure, done.

File: source/simulator/stokes_matrix_free.cc [hunk 2]
```diff
@@ -2396,14 +2396,13 @@ namespace aspect
     }
 
     // Setup the matrix-free operators
-    std::shared_ptr<MatrixFree<dim,double>> stokes_mf_storage(new MatrixFree<dim,double>());
-    matrix_free_objects.push_back(stokes_mf_storage);
+    std::shared_ptr<MatrixFree<dim,double>> matrx_free = std::make_shared<MatrixFree<dim,double>>();
```
Comment (MFraters) [inline source/simulator/stokes_matrix_free.cc:2399]: Can you fix `matrx_free` to `matrix_free`? Otherwise good to merge.

Comment (tjhei) [reply source/simulator/stokes_matrix_free.cc]: That would help, right? Haha.
